### PR TITLE
Create tabletransforms.md

### PR DESCRIPTION
**Pull request by juliohm**
Appreciate if you can review this proposal. I am happy to mentor beginners or intermediate level students interested in ML and data science. These transforms can become quite useful to the Julia ML ecosystem in the near future.

### CODE REVIEW

**Comment by logankilpatrick** [comment]
Hey, can you add how long each project is [175 hrs or 350 hrs] and then also add a link to this file in the GSoC index so folks can find these?

File: jsoc/gsoc/tabletransforms.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Statistical transforms
+
+Statistical transforms such as PCA, Z-score, etc, can greatly improve the convergence of various statistical
+learning models, and are widely used in advanced machine learning pipelines. In this project the mentee will
+learn how to implement advanced transforms such as PPMT and other transforms for imputation of missing values.
+
+**Desired skills:** Statistics, Machine Learning
+
+**Difficulty level:** Medium
```
Comment (juliohm) [inline jsoc/gsoc/tabletransforms.md:17]:
```suggestion
**Difficulty level:** Medium

**Expected duration:** 350hrs
```

File: jsoc/gsoc/tabletransforms.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Utility transforms
+
+Utility transforms such as standardization of column names and other string-based transforms are extremely important
+for digesting real-world data. In this project the mentee will learn good coding practices and will implement various
+utility transforms available in other languages (e.g. Janitor package in R, pyjanitor in Python).
+
+**Desired skills:** Text processing, Regex
+
+**Difficulty level:** Easy
```
Comment (juliohm) [inline jsoc/gsoc/tabletransforms.md:34]:
```suggestion
**Difficulty level:** Easy

**Expected duration:** 175hrs
```

**Comment by juliohm** [comment]
Thank you @logankilpatrick , I've added the time duration for each project. Is there a link explaining more details about the format this year? Dates, expected hours per day, etc.

**Comment by juliohm** [comment]
Can you point out where exactly I should add the link to this proposal file? I looked quickly into the repository and the index.md file in the parent directory doesn't seem to contain links to files in this directory.

**Comment by logankilpatrick** [comment]
This file: https://github.com/JuliaLang/www.julialang.org/blob/main/jsoc/projects.md is where the link needs to go and here is the blog post with details on what makes GSoC different this year: https://opensource.googleblog.com/2021/11/expanding-google-summer-of-code-in-2022.html

**Comment by juliohm** [comment]
Done @logankilpatrick 👍🏽 First time I used the built-in VSCode in GitHub to edit a PR. We can just press "." and edit the files in the PR as if it we had a local VSCode session. 💯

**Comment by logankilpatrick** [comment]
Awesome, thank you for this!

**Pull request merged**